src/libs/actions/App.js

```diff
@@ -141,30 +141,52 @@ function getPolicyParamsForOpenOrReconnect() {
 
 /**
  * Returns the Onyx data that is used for both the OpenApp and ReconnectApp API commands.
+ * @param {Boolean} isOpenApp
  * @returns {Object}
  */
-function getOnyxDataForOpenOrReconnect() {
+function getOnyxDataForOpenOrReconnect(isOpenApp = false) {
     return {
         optimisticData: [
             {
                 onyxMethod: Onyx.METHOD.MERGE,
                 key: ONYXKEYS.IS_LOADING_REPORT_DATA,
                 value: true,
             },
+            isOpenApp
```

Comment:
instead of making if else logic inside object - i will suggest to add one default one
```
const defaultData = {
    optimisticData: [
        {
            onyxMethod: 1,
            key: 2,
            value: true,
        },
    ],
    successData: [
        {
            onyxMethod: 3,
            key: 4,
            value: false,
        },
    ],
    failureData: [
        {
            onyxMethod: 5,
            key: 6,
            value: false,
        },
    ],
};
```
And add condition at line 148` if (!isOpenApp) return defaultData`
And after it
```
return {
    ...defaultData,
    optimisticData: [...defaultData.optimisticData, {
        onyxMethod: 3,
        key: ONYXKEYS.IS_LOADING_APP,
        value: true,
    }],
    successData: [...defaultData.successData, {
        onyxMethod: 3,
        key: ONYXKEYS.IS_LOADING_APP,
        value: true,
    }],
    failureData: [...defaultData.failureData, {
        onyxMethod: 3,
        key: ONYXKEYS.IS_LOADING_APP,
        value: true,
    }]
}
```